cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -976,7 +976,7 @@ private boolean isGettingMoreBalanced(Broker sourceBroker, double sourceUtilizat
     double prevDiff = (sourceBrokerUtilization / sourceBrokerCapacity) - (destinationBrokerUtilization / destinationBrokerCapacity);
     double nextDiff = prevDiff + (sourceUtilizationDelta / sourceBrokerCapacity) + (sourceUtilizationDelta / destinationBrokerCapacity);
 
-    return Math.abs(nextDiff) < Math.abs(prevDiff);
+    return Math.abs(nextDiff) <= Math.abs(prevDiff);
```

Comment:
Couldn't this introduce a side effect in the general case where big partitions of the same size could be swapped since the diff would be the same? or is the change fully deterministic?

Reply:
Hmm, good question.

Had a dig through the code, and on the case of partition swaps, that case is already handled specially. When `sourceUtilizationDelta` is zero (which, for replica swaps, means the partitions are the same size), it's _already_ being special-cased to return `ActionAcceptance.ACCEPT` before we even call this function ([code](https://github.com/DataDog/cruise-control/blob/14f48dcb7260bc8e265172ae2cbc3fa57f742769/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java#L113)). I think that special-casing would be unnecessary with this change.

I think for most users returning `ActionAcceptance.ACCEPT` in the swap case should generally be fine. Because of our large partitions have a problem with resource usage going high in the middle of the reassignment which we think is mostly due to swaps, @kphelps is working on a custom goal to `REJECT` those when they look dangerous.